recipe/recipe.go

```diff
@@ -0,0 +1,214 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/JulzDiverse/cfclient"
+	"github.com/julz/cube"
+	"github.com/pkg/errors"
+	"github.com/starkandwayne/goutils/ansi"
+
+	"code.cloudfoundry.org/bbs/models"
+	"code.cloudfoundry.org/runtimeschema/cc_messages"
+)
+
+func main() {
+	downloadUrl := os.Getenv(cube.EnvDownloadUrl)
+	uploadUrl := os.Getenv(cube.EnvUploadUrl)
+	appId := os.Getenv(cube.EnvAppId)
+	stagingGuid := os.Getenv(cube.EnvStagingGuid)
+	completionCallback := os.Getenv(cube.EnvCompletionCallback)
+
+	username := os.Getenv(cube.EnvCfUsername)
```

Comment:
do we need to think about the security of this? if this is a cf admin username/password we probably don't want it inside a container running user code?

Reply:
Yes that is also an TODO. The credentials are required to download/upload bits/dorplets. 
- We should make sure to delete jobs after they are done (unfortunately there is no way to do it using the manifest of a job)
- Runt tasks in a separate space
- Kubernetes has something for storing credentials afaik, so we could have those credentials in kube instead of providing it when running a task.

Reply:
https://github.com/julz/cube/issues/15